src/rat/ee_utils/ee_aec_file_creator.py

```diff
@@ -24,67 +28,479 @@ def grouper(iterable, n, *, incomplete='fill', fillvalue=None):
-  print("AEC file exists for all reservoirs in this basin")
-  return 1
+                ## Preparing dataframe to write down to csv 
+        aec_df = pd.DataFrame(
+                    data = {
+                        'Elevation' :elevs_list, 
+                        'CumArea':areas_list
+                        })
+        print(f"Observed AEC obtained from SRTM for {reservoir_name}")
+
+    if clip_to_water_surf:
+        aec_df = get_obs_aec_above_water_surface(
+            aec_df
+        )
+        print(f"Clipped to elevations above water surface")
+
+    aec_df.to_csv(aec_dst_fp,index=False)
+
+    return aec_df
+
+def aec_file_creator(
+        reservoir_shpfile, shpfile_column_dict, aec_dir_path, 
+        scale=30, dam_bottom_elevation_percentile=1, 
+        dam_buffer_distance=500, 
+        grwl_fp=None
+    ):
+    """
+    Creates AEC (Area-Elevation Curve) files for reservoirs based on the provided shapefile.
+
+    Parameters:
+    - reservoir_shpfile (str or gpd.geodataframe.GeoDataFrame): Path to the reservoir shapefile or a GeoDataFrame.
+    - shpfile_column_dict (dict): Dictionary mapping shapefile columns to their respective data types. 
+      Required keys:
+        - 'unique_identifier': Column name for the unique identifier of the reservoir.
+        - 'dam_height': Column name for the height of the dam.
+        - 'dam_lat': Column name for the latitude of the dam.
+        - 'dam_lon': Column name for the longitude of the dam.
+    - aec_dir_path (str): Directory path where AEC files will be stored.
+    - scale (int, optional): Scale for the AEC calculation. Default is 30.
+    - dam_bottom_elevation_percentile (int, optional): Percentile for dam bottom elevation. Default is 1.
+    - dam_buffer_distance (int, optional): Buffer distance around the dam. Default is 500.
+    - grwl_fp (str, optional): File path to the GRWL (Global River Widths from Landsat) dataset. Default is None. 
+        Can be passed as 'grwl' option in 'GLOBAL' section of config file.
+    """
+    # Obtaining list of csv files in aec_dir_path
+    aec_filenames = []
+    for f in os.listdir(aec_dir_path):
+        if f.endswith(".csv"):
+            aec_filenames.append(f[:-4])
+    if isinstance(reservoir_shpfile, str):
+        reservoirs_polygon = gpd.read_file(reservoir_shpfile)
+    elif isinstance(reservoir_shpfile, gpd.geodataframe.GeoDataFrame): 
+        reservoirs_polygon = reservoir_shpfile
+    else:
+        raise ValueError("reservoir_shpfile should be either a string or a geodataframe")
+    for reservoir_no,reservoir in reservoirs_polygon.iterrows():
+        # Reading reservoir information
+        reservoir_gpd = gpd.GeoSeries(reservoir.geometry)
+        reservoir_gpd = reservoir_gpd.set_crs(reservoirs_polygon.crs)
+
+        reservoir_name = str(reservoir[shpfile_column_dict['unique_identifier']])
+        dam_height = float(reservoir[shpfile_column_dict['dam_height']])
```

Comment:
We need to handle the case where dam_height is NaN or negative like (-99). Worst case is if there is no dam_height, return strm observed aec without any extrapolation.